sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -60,6 +66,35 @@
     private final ReactorHandlerProvider handlerProvider;
     private final Mono<ClaimsBasedSecurityNode> cbsNodeSupplier;
 
+    private final AtomicReference<LinkSubscription<AmqpLink>> coordinator = new AtomicReference<>();
+
+    private AmqpRetryPolicy retryPolicy;
+
+    /**
+     *
+     * Creates a new AMQP session using proton-j.
+     *
+     * @param session Proton-j session for this AMQP session.
+     * @param sessionHandler Handler for events that occur in the session.
+     * @param sessionName Name of the session.
+     * @param provider Provides reactor instances for messages to sent with.
+     * @param handlerProvider Providers reactor handlers for listening to proton-j reactor events.
+     * @param cbsNodeSupplier Mono that returns a reference to the {@link ClaimsBasedSecurityNode}.
+     * @param tokenManagerProvider Provides {@link TokenManager} that authorizes the client when performing
+     *     operations on the message broker.
+     * @param retryPolicy for the session operation to complete.
+     */
+    public ReactorSession(Session session, SessionHandler sessionHandler, String sessionName, ReactorProvider provider,
+                          ReactorHandlerProvider handlerProvider, Mono<ClaimsBasedSecurityNode> cbsNodeSupplier,
```

Comment:
Why do we need this constructor overload if we can just add a retry to the existing ctor.